include/material_properties.tcc

```diff
@@ -0,0 +1,6 @@
+//! Constructor with density and k0
+//! \param[in] json input file containing material properties
+MaterialProperties::MaterialProperties(const json& jsonfile) {
+  density_ = jsonfile["material_properties"]["density"].get<double>();
```

Comment:
This should already be the `material_properties` object. This class shouldn't have access to other properties in JSON.